### PR TITLE
Dimension store

### PR DESCRIPTION
**Pull request by laureleya**
Adds the dimension store and tests for the same 
(not finished because final implementation is dependent on dimension ids from the backend)

### CODE REVIEW

**Review by keli95566 — APPROVED** [review]
In general, very very nice job of implementing the dimension store on your own! I have made a few suggestions which could make dimension actions / functions easier to use for the future, and a few things to make things a bit cleaner. But those changes could also be made in the future.  

P.s. The tests are failing because of my previous changes in the FieldItem from "npm run test" -.-... I will change the way we tested FieldItem before and this could be fixed. Nothing from your PR that caused the failed test.

File: src/components/ListModal/ListModalContainer.js [hunk 1]
```diff
@@ -13,7 +13,9 @@ import {
   createNewVisualization,
   fillVisualization,
 } from "../../states/visualizations/actions";
-import { createNewModel } from "../../states/models/actions";
+import { createNewModel, updateModelDimensions } from "../../states/models/actions";
```
[the remark this answers is not in the document]
Comment (laureleya) [reply src/components/ListModal/ListModalContainer.js:16]: Yeah, that would be great!

File: src/components/ListModal/ListModalContainer.js [hunk 2]
```diff
@@ -49,7 +53,14 @@ class ListModalContainer extends React.Component {
         fetchModelData(item)
           .then((response) => {
             createNewModel(item, response["Fields"]);
+            return JSON.parse(JSON.stringify(response["Fields"]));
           })
+          .then((value) => {
+            addAllDimensions(this.props.modelId, item, value);
+          }).then(() => {
+            updateModelDimensions(this.props.modelId,  this.props.getDimensionsOfCurrentModel);
```
[the remark this answers is not in the document]
Comment (laureleya) [reply src/components/ListModal/ListModalContainer.js:61]: I ran into the problem that, when put into the models store reducer, I cannot access the state I need for the getDimensionsOfCurrentModel. But I can very well put it in front of it.

File: src/components/ListModal/ListModalContainer.js [hunk 3]
```diff
@@ -93,10 +104,12 @@ class ListModalContainer extends React.Component {
 
 const mapStateToProps = (state) => {
   return {
+    dimensions: state.dimensions,
```
[the remark this answers is not in the document]
Comment (laureleya) [reply src/components/ListModal/ListModalContainer.js:107]: Nothing, it was from an earlier version. I'll delete it.

File: src/states/dimensions/reducer.js [hunk 4]
```diff
@@ -1,19 +1,72 @@
-import { CREATE_NEW_DIMENSION } from "./constants";
+import { ADD_ALL_DIMENSIONS} from "./constants";
+import update from "immutability-helper";
+import { v4 as uuidv4 } from 'uuid';
 
-const defaultState = {
-  dimensions: {}
+export const defaultState = {
+  dimensions: {
+    byDimensionName: {},
```
[the remark this answers is not in the document]
Comment (laureleya) [reply src/states/dimensions/reducer.js:7]: I wanted to make it consistent with the model store where there is also models.models.

**Pull request merged**